libevmcore/Assembly.cpp

```diff
@@ -171,48 +173,74 @@ ostream& dev::eth::operator<<(ostream& _out, AssemblyItemsConstRef _i)
 	return _out;
 }
 
-ostream& Assembly::streamRLP(ostream& _out, string const& _prefix) const
+string Assembly::getLocationFromSources(StringMap const& _sourceCodes, SourceLocation const& _location) const
+{
+	if (_location.isEmpty() || _sourceCodes.empty() || _location.start >= _location.end || _location.start < 0)
+		return "";
+
+	auto it = _sourceCodes.find(*_location.sourceName);
+	if (it == _sourceCodes.end())
+		return "";
+
+	string toReturn = it->second;
+	if (_location.start >= (int)toReturn.size())
+		return "";
+
+	toReturn = toReturn.substr(_location.start, _location.end - _location.start);
+	auto newLinePos = toReturn.find_first_of("\n");
+	if (newLinePos != string::npos && newLinePos != toReturn.size() - 1)
```

Comment:
I think this condition is still wrong. If the last character is a newline, it will be included in `toReturn`.